src/Microsoft.AspNetCore.Razor.Runtime/Resources.resx

```diff
@@ -195,4 +195,16 @@
   <data name="ArgumentMustBeAnInstanceOf" xml:space="preserve">
     <value>Argument must be an instance of '{0}'.</value>
   </data>
+  <data name="TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace" xml:space="preserve">
+    <value>Could not find matching ']' for required attribute.</value>
+  </data>
+  <data name="TagHelperDescriptorFactory_InvalidRequiredAttributeCharacter" xml:space="preserve">
+    <value>Invalid required attribute character '{0}'. Separate required attributes with commas.</value>
+  </data>
+  <data name="TagHelperDescriptorFactory_UnexpectedEndOfRequiredAttribute" xml:space="preserve">
+    <value>Reached the end of the required attribute unexpectedly. Separate required attributes with commas.</value>
```

Comment:
Suggest dropping the second sentence because it only sometimes applies. Neither of these two messages have any definite relationship to how the `Attributes` value is split up. A semicolon between bare attribute names is just one of many possible issues.